sklearn/utils/testing.py

```diff
@@ -697,5 +706,34 @@ def check_skip_travis():
     if os.environ.get('TRAVIS') == "true":
         raise SkipTest("This test needs to be skipped on Travis")
 
+
+def delete_folder(folder_path):
+    """Utility function to cleanup a temporary folder if still existing. Copy from joblib.pool (for independance)"""
+    if os.path.exists(folder_path):
+        shutil.rmtree(folder_path)
+
+
+class TempMemmap(object):
+    def __init__(self, X, mmap_mode='r'):
```

Comment:
As X could be anything (not just a 2D datastructure, it could be a typle of arrays for instance) I would instead use a more generic name such as `data`.

In sklearn capital letters for variable names are reserved for 2D datastructures (2D numpy arrays or scipy sparse matrices).